### DISCUSSION TITLE
Request: 'odometer', 'mileage', and-or related entities with a recommended design pattern

### DISCUSSION THREAD

**Discussion by jonathanvajda** [CommonCoreOntology/CommonCoreOntologies#390] [Announcements]
Use cases:
- a fleet of vehicles have been used at various levels, and the primary way of tracking use has been following mileage for each vehicle, and reporting daily and weekly.
- a specific vehicle travels 25 kilometers on Tuesday. The odometer has value "10000" in the morning, and the odometer has value "10025" in the evening.

CCO already has:
- Measurement Unit of Length (class)
- Mile Measurement unit (instance)
- Kilometer Measurement unit (instance)
- Recording device (class)
- Instrument display panel (class)
- Distance Measurement Information Content Entity

Now, I am not sure what would be the parent class exactly for 'odometer' or 'mileage'. But it seems that what I am looking for is an information bearing artifact and uses_measurement_unit 'Mile Measurement Unit' (or in the second case, 'Kilometer Measurement Unit'). But what does it measure/what is the measurement of? It seems to me that the measurement is about distance or length, not of some material object but rather some spatio(temporal) path that the vehicle has passed through.

**Could we get a recommended design pattern for these cases?**

In the modeling information in CCO v1.3 documentation it gives recommendations for speed, specified function ranges, etc., but I may be missing where it recommends measuring distance _traveled_ per se.

**Comment by cameronmore** [comment]
I think this is a good point, and I think we should add odometer as an IBA. It means roughly in Greek 'the measurement of a path,' so although it works using time, the spinning of wheels, etc, it's ultimately a measurement of the distance traveled. Different odometers use different methods for calculating that distance, but the distance is the core measurement the artifact is designed to capture. I'm not sure about committing to the temporal part of spatio-temporal.

Draft definition: "An Odometer is an Information Bearing Artifact that is designed to bear some Information Content Entity that measures the distance an entity has traveled." Should we add that it measures the distance of the entity it is attached to? Should we say anything about the means by which is measures? Not sure, happy to start the conversation.

An odometer definitely bears a Measurement Artifact Function, but does the definition of Distance Measurement Artifact Function fit? "A Measurement Artifact Function that is realized during events in which an Artifact is used to measure the spatial Distance to a specified object or class of objects."

If this issue gets discussion, I would be happy to advocate this to be added to the upcoming 1.5 release with the changes to the relations.

I'll think further on current design patterns recommendations in lieu of this term.

**Comment by jonathanvajda** [comment]
>Draft definition: "An Odometer is an Information Bearing Artifact that is designed to bear some Information Content Entity that measures the distance an entity has traveled."

Good enough, but "distance an entity has traveled" could reuse terms in the CCO lexicon, like "distance of some spatial region that was the location of some Act of Location Change"? maybe that's too clunky.

I take it a car's odometer would be paradigmatic. Would someone say that our definition includes pedometers for lack of specific differences to distinguish the two?

>Should we say anything about the means by which is measures? Not sure, happy to start the conversation

I would imagine that we'd want to leave it open what measurement unit is and what was the mechanism for deriving its measure. I don't need it to be more specific. What do you have in mind?

>An odometer definitely bears a Measurement Artifact Function, but does the definition of Distance Measurement Artifact Function fit? "A Measurement Artifact Function that is realized during events in which an Artifact is used to measure the spatial Distance to a specified object or class of objects."

I think so. Do you hesitate?

>I'll think further on current design patterns recommendations in lieu of this term.

I am tempted to draw a diagram, but I also am somewhat unsure.

Related point: is the mileage just the ICE? Or is the mileage what the ICE measures? I am getting ambiguity in definitions online. Some say it is the calculation or it is the number, whereas others describe it as the distance or path. I would think that mileage is what is _measured_. If I am correct, then, an odometer measures _mileage (or kilometer equivalent) traveled_, and mileage is _distance along a path_.

**Comment by cameronmore** [comment]
> > Draft definition: "An Odometer is an Information Bearing Artifact that is designed to bear some Information Content Entity that measures the distance an entity has traveled."

> Good enough, but "distance an entity has traveled" could reuse terms in the CCO lexicon, like "distance of some spatial region that was the location of some Act of Location Change"? maybe that's too clunky.

We could use that or something similar, that there's a path way, or 2d or 3d region that the car traveled. Do we want to restrict odometers to vehicles and not pedometers, maybe a pedometer is more specific, a subclass of odometer? We could just say odometers measure bfo:objects in the definition and call it there.

> Related point: is the mileage just the ICE? Or is the mileage what the ICE measures? I am getting ambiguity in definitions online. Some say it is the calculation or it is the number, whereas others describe it as the distance or path. I would think that mileage is what is _measured_. If I am correct, then, an odometer measures _mileage (or kilometer equivalent) traveled_, and mileage is _distance along a path_.

Seems like mileage is the value of the measurement. We don't say 25-mileage as if mileage is a unit, but it's the thing that the measurement value _is_. 25 is the 1025 is the mileage which is measured by the unit 'mile'.
